### PR TITLE
Add ability to enable/disable the extended kalman filter (EKF)

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Add ability to toggle the extended kalman filter in navigation native.

- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

We want to be able to able to toggle the extended kalman filter path.

### Implementation

Mapbox location native has a [simple boolean check](https://github.com/mapbox/mapbox-location-native/blob/64b59dee47b47066535815b43517d01a078ec437/src/mblc/mapbox_location.cpp#L209-L220) to see if the configuration supports EKF. 

This will make it so we can enable/disable EKF at any point in the app lifecycle.

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

cc: @Guardiola31337 @mskurydin @LukasPaczos

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -529,6 +529,10 @@ constructor(
         tripSession.updateSensorEvent(sensorEvent)
     }
 
+    fun useExtendedKalmanFilter(useEKF: Boolean) {
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: Could you add the docs here as well?

**Comment by Guardiola31337** [comment]
> Going forward we should come up with a strategy for mutable options. Should everything exposed in `NavigationOptions` be mutable? We could possibly be able to reset all available there options and pass them to interested modules with a setter.

Capturing from @SiarheiFedartsou 
> As I can see from code we can change it any time(even between location updates, but I wouldn't recommend to do it)

It may work for this specific option but not for all of them. This question was brought in the past and we decided to only set it at construction time and maintain it during `MapboxNavigation` lifetime 👀 how it's implemented in the legacy code https://github.com/mapbox/mapbox-navigation-android/blob/d4aa4cfd46a2c69d2cc60366bbd508ab5a9a86b7/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L1034 https://github.com/mapbox/mapbox-navigation-android/blob/d4aa4cfd46a2c69d2cc60366bbd508ab5a9a86b7/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L1077-L1085

We should work with @mapbox/navnative and see what options we may want to expose and if they can be updated on the fly.

On a different topic, I'm not a big fan of this solution as `MapboxNavigation` will keep growing and growing as we expose new options. I believe we should come up with a different approach that don't clutter the entry point of the SDK.

**Comment by SiarheiFedartsou** [comment]
I think the best solution is to think about `setConfig` as if it is a part of constructor. I believe at some point we will move it to constructor as well as `configureRouter`.

**Comment by LukasPaczos** [comment]
> I think the best solution is to think about setConfig as if it is a part of constructor. I believe at some point we will move it to constructor as well as configureRouter.

This would mean that the implementation from this PR is already semi-deprecated and will require breaking changes in the near future to remove the introduced here mutability.

In this case, I think we should refactor this option and pass it only in the `NavigationOption` when the `MapboxNavigation` is first created.

**Pull request merged**